code/game/objects/items/weapons/RPD.dm

```diff
@@ -0,0 +1,181 @@
+//Contains the rapid piping device. Copied over largely from pipe_dispenser and RCD files.
+//BUG: RPD uses charge when clicking on something, even if it doesn't create pipes.
+/obj/item/weapon/rpd
+	name = "rapid piping device"
+	desc = "A device used to rapidly build pipes."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcd" //Same as RCD, currentely.
+	opacity = 0
+	density = 0
+	anchored = 0.0
+	flags = CONDUCT
+	force = WEAPON_FORCE_PAINFUL
+	throwforce = WEAPON_FORCE_PAINFUL
+	throw_speed = 1
+	throw_range = 5
+	w_class = ITEM_SIZE_NORMAL
+	origin_tech = list(TECH_ENGINEERING = 2, TECH_MATERIAL = 2) //Redundant values that yer free ta' change later.
+	matter = list(MATERIAL_STEEL = 10, MATERIAL_PlASMA = 2, MATERIAL_PLASTIC = 5) //Redundant values that yer free ta' change later.
+	price_tag = 500 //Redundant values that yer free ta' change later.
+	var/use_power_cost = 1.5
+	var/obj/item/weapon/cell/cell = null
+	var/suitable_cell = /obj/item/weapon/cell/medium
+	var/wait = 0
+	var/pipe_type = 0
+	var/p_dir = 1
+
+/obj/item/weapon/rpd/attack()
+	return 0
+
+/obj/item/weapon/rpd/proc/can_use(var/mob/user,var/turf/T)
+	return (user.Adjacent(T) && user.get_active_hand() == src && !user.stat && !user.restrained())
+
+/obj/item/weapon/rpd/MouseDrop(over_object)
+	if((src.loc == usr) && istype(over_object, /obj/screen/inventory/hand) && eject_item(cell, usr))
+		cell = null
+	else
+		return ..()
+
+/obj/item/weapon/rpd/attackby(obj/item/C, mob/living/user, obj/item/I)
+	if(istype(C, suitable_cell) && !cell && insert_item(C, user))
+		src.cell = C
+	return
+
+/obj/item/weapon/rpd/proc/useCharge()
+	//Use charge from the cell.
+	if(!cell||!cell.check_charge(use_power_cost))
```

Comment:
You can cell.check_use instead of cell.check_charge, and it'll consume the power from the cell (and make it explode if it's rigged)

Reply:
So replace cell.check_charge with cell.check_use?

Reply:
Or do ye mean replace with cell.checked_use?

Reply:
That did seem ta' help with the charge being used each click bug if that's what you meant.